src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -776,6 +777,18 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
                 return false;
             }
             ((CreateIndexRequest) request).index(newIndices.length != 1 ? null : newIndices[0]);
+        } else if (request instanceof ResizeRequest) {
+            // clone or shrink operations
```

Comment:
why is this change needed? Is it only for testing?

Reply:
I noticed that test shrinkIndex_positive()  and others that dealt with shrinking index were ignored from the start. The request that was created in test wasn't analyzed in any way in indexResolver. After this change request is checked, authorized against index and if user has permissions, shrink operation takes place. 
I'm not sure yet if this is the case in manual testing, have to try it

Reply:
okay, you can post the results of your manual testing here. I know you were planning to address this separately. Maybe file an issue and a follow-up PR?

Reply:
I was able to reprocude the error by manual testing, also found old task documenting the issue. Here's the task: https://github.com/opensearch-project/security/issues/2141
Here's PR where I cherry picked fix changes from this branch: https://github.com/opensearch-project/security/pull/3716
All shrink operation/resize tests are un-ignored and are passing, I'll cleanup this branch to not push same changes 2 times.